### DISCUSSION TITLE
AuthnBackend / AuthzBackend

### DISCUSSION THREAD

**Discussion by DougLau** [maxcountryman/axum-login#180] [General]
Being new to this library, the `AuthnBackend` and `AuthzBackend` trait names are a bit confusing.  Differing by only one lower-case letter, and containing the overloaded term "Backend" had me scratching my head.  Perhaps it's too late to change, but `Authenticate` / `Authorize` would make more sense to my brain.

**Comment by maxcountryman** [comment]
The concept of an authentication backend is [borrowed from Django](https://docs.djangoproject.com/en/5.0/topics/auth/customizing/#customizing-authentication-in-django) which is the primary source for this crate's design.

Authentication backends are the core concept central the way axum-login works and for that reason we want to ensure the trait names reflect that: When you implement `AuthnBackend` and `AuthzBackend`, you are implementing the conceptual notion of what we mean by an authentication backend.

In our case we have both an "authenticating" backend, that is a backend which provides authentication and an "authorizing" backend, a backend providing authorization. This split exists because it's easier to model the fact that not all authn backends will also support authz.

Of course everyone's experience will vary here, and it's important to bear in mind that the curse of knowledge often complicates things further, but I've found the convention of using "authn" and "authz" to indicate authentication and authorization respectively to be ubiquitous, well known, and most importantly clear. That said, perhaps there are ways we could improve the documentation to make it less confusing for folks for whom this is their first contact with those terms?

**Reply by DougLau** [reply]
Thanks, that makes sense.  I'm sure I'll get used to it.

**Reply by HosMercury** [reply]
should not backend called Connect bc it contains the connections ? better?
